eval/src/vespa/eval/eval/value_cache/constant_tensor_loader.cpp

```diff
@@ -76,23 +79,33 @@ ConstantTensorLoader::create(const vespalib::string &path, const vespalib::strin
         LOG(warning, "invalid type specification: %s", type.c_str());
         return std::make_unique<SimpleConstantValue>(_engine.from_spec(TensorSpec("double")));
     }
-    Slime slime;
-    decode_json(path, slime);
-    std::set<vespalib::string> indexed;
-    for (const auto &dimension: value_type.dimensions()) {
-        if (dimension.is_indexed()) {
-            indexed.insert(dimension.name);
-        }
+    if (ends_with(path, ".tbf")) {
+        vespalib::File file(path);
+        file.open(File::READONLY);
+    	std::vector<char> content(file.stat()._size);
+	file.read(&content[0], content.size(), 0);
+        vespalib::nbostream_longlivedbuf stream(&content[0], content.size());
+        return std::make_unique<SimpleConstantValue>(_engine.decode(stream));
     }
-    TensorSpec spec(type);
-    const Inspector &cells = slime.get()["cells"];
-    for (size_t i = 0; i < cells.entries(); ++i) {
-        TensorSpec::Address address;
-        AddressExtractor extractor(indexed, address);
-        cells[i]["address"].traverse(extractor);
-        spec.add(address, cells[i]["value"].asDouble());
+    else {
```

Comment:
no need to use else here
and at least not like this (bracing)

Reply:
No need, but clearer in my opinion because it makes it clear the two blocks are alternatives at the same footing.
If the else block is promoted outside the block it looks like the default option while the if block looks to me like an escape for a special case (like the if block above).

Anyway, it's you code. Let me know if it is the second option you prefer and I'll change it.

Reply:
Done.